src/components/layout/Section.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+
+export const VALID_BG_NORMAL = ['white', 'haze', 'lightGray'];
+
+export const VALID_BG_INVERTED = ['navy'];
+
+export const VALID_PADDING = ['default', 'double', 'half', 'none'];
+
+/**
+ * @param {String} axis 'h' or 'v'
+ * @param {String} amount
+ * @returns {String} padding utility classes
+ */
+export const getPaddingClasses = (axis, amount) => {
+  const sides = axis === 'h' ? ['left', 'right'] : ['top', 'bottom'];
+
+  let classes = `padding--${sides[0]} padding--${sides[1]}`;
+  if (amount !== 'default') {
+    classes = `padding--${sides[0]}--${amount} padding--${sides[1]}--${amount}`;
+  }
+
+  return classes;
+};
+
+/**
+ * @param {String} direction
+ * @returns {String} border utility classes
+ */
+export const getBorderClasses = (direction) => {
+  let classes = '';
+  switch (direction) {
+    case 'h':
+      classes = 'border--left border--right';
+      break;
+    case 'v':
+      classes = 'border--top border--bottom';
+      break;
+    default:
+      classes = `border--${direction}`;
+  }
+  return classes;
+};
+
+/**
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const Section = ({ hPadding, vPadding, bgColor, border, children }) => {
```

Comment:
This is something that might be interesting to talk about. What components should use rest props, vs which ones shouldn't, and why. I'm personally not sure if I'd want to do the thing where a developer has to ask for data attribute or whatever first, for us to add it... I feel like we should either support rest props, or support nothing, as we're doing here.

Reply:
`data-test` is the best reason I can think of with supporting rest props, and I don't think `Section` is the right place to put that attribute anyway.

I'm of the opinion that the webdriver should be hooking into specific content inside layout components, not the layout components themselves.

Reply:
Talked offline. Layout components shouldn't get rest props (and this is a layout component).